polygon/sync/service.go

```diff
@@ -33,6 +33,7 @@ import (
 )
 
 type Service interface {
+	HeimdallService() heimdall.Service
```

Comment:
The sync service is not responsible for "providing" the heimdal.Service. They are just 2 components that work on their own domain. I would suggest to do what we did with the Bridge integration - ie instantiate heimdall service outside of sync service and pass it to both sync service and bor consensus engine

Reply:
Sounds good, I had a similar thought when I looked at the diff - I'll do this instead. 

Maybe we should have a helper function (maybe in `backend.go` or somewhere in the `polygon` package) that instantiates the bridge and the heimdall service as well as any future components that are accessed externally.